### ISSUE TITLE
Suspected memory leak issue in the code

### ISSUE THREAD

**Issue by Aceticia** [lukecavabarrett/pna#4]
Hello, I have an existing project that uses different types of graph neural networks. Other neural networks from the official dgl implementations are working normally, but the dgl version of your code experiences GPU memory leaks. Which version of dgl are you using?

**Comment by Aceticia** [comment]
This is the definition for the net:
![image](https://user-images.githubusercontent.com/36082433/92556191-4d2ae000-f237-11ea-816e-44ce98368547.png)

And this is how I call it, 
`
class_h = self.class_layers(g=g, h=raw_feat, e=g.edata['weight'].detach())
`
where g is a dgl graph. What might be the potential cause of this? Any suggestion would be helpful, thanks in advance.
